### ISSUE TITLE
Shadows get clipped for large critters

### ISSUE THREAD

**Issue by DudeMcDude** [GrognardsFromHell/TemplePlus#195]
Example: Hezrou Guardian (proto 14360). Might have to play its attack animation to see it.

**Comment by shartte** [comment]
In which shadow mode? Shadow maps?

**Comment by DudeMcDude** [comment]
Yeah shadow maps.

**Comment by shartte** [comment]
Have you checked whether this bug is present in vanilla?

**Comment by shartte** [comment]
A bit of background info: The positioning of the shadowmap texture on the ground is based on the object radius and object render height. Both are calculated "inaccurately" by taking the current animation frame and calculating the height of that. That is only done once however, for the idle animation. So if a model has an extremely high attack animation, it might exceed the calculated size for the shadow map buffer.

I wonder if this is already a problem with the vanilla approach. I'll try to reproduce this in vanilla.

**Comment by shartte** [comment]
Checked against the Hezrou Guardian in Vanilla (co8):
Radius = 45
Height = 152.79

I verified that TemplePlus is using the same values.

**Comment by shartte** [comment]
Here's a screenshot of the issue on TemplePlus:
![2016-04-03 13_12_35-temple of elemental evil temple](https://cloud.githubusercontent.com/assets/1261399/14232119/c28a7140-f99d-11e5-9373-b2127d21a808.png)

**Comment by shartte** [comment]
This is how it looks in vanilla, which also doesn't seem to be correct:
![2016-04-03 13_29_21-toeea exe](https://cloud.githubusercontent.com/assets/1261399/14232166/1ec2942c-f9a0-11e5-8624-671491ec8aa0.png)

**Comment by shartte** [comment]
I suggest we put this to lower priority since it seems to be present in vanilla too.

This is actually not that easy to fix with the strange shadow map algorithm that ToEE uses.

**Comment by DudeMcDude** [comment]
Yeah, I guess I never noticed it now since I always played with blob shadows.
